pkg/shared/util/grpc_utils.go

```diff
@@ -20,18 +20,98 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"os"
+	"reflect"
+	"strings"
 	"time"
 
+	"github.com/Masterminds/semver/v3"
+	pep440 "github.com/aquasecurity/go-pep440-version"
 	"github.com/numaproj/numaflow-go/pkg/info"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/numaproj/numaflow"
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+func checkConstraint(version *semver.Version, constraint string) error {
+	c, err := semver.NewConstraint(constraint)
+	if err != nil {
+		return fmt.Errorf("error parsing constraint: %w", err)
+	}
+
+	isValid, _ := c.Validate(version)
+	if !isValid {
+		return fmt.Errorf("version did not meet constraint requirement")
+	}
+
+	return nil
+}
+
+// checkCompatibility checks if the current numaflow version is compatible with the given language's SDK version
+func checkCompatibility(serverInfo *info.ServerInfo, versionMappingConfig map[string]sdkConstraints, numaflowVersion string) error {
+	// If we are in CI or testing locally, we do not need to check for compatibility issues
+	if !strings.Contains(numaflowVersion, "latest") && os.Getenv("GITHUB_ACTIONS") == "true" {
+		print("keran is testing, the version is " + numaflowVersion)
+		// Check if server info contains the MinimumNumaflowVersion field
+		sdkVersion := serverInfo.Version
+		rv := reflect.ValueOf(serverInfo)
```

Comment:
What is the reason we can not use `serverInfo.minimumNumaflowVersion` to compare?

Reply:
If the user is using an old version of SDK, the ServerInfo struct might not have the field just yet.

Reply:
In that case, won't the `minimumNumaflowVersion` be empty string after parsing?

Reply:
@ayildirim21

Reply:
if you try accessing a field that doesnt exist in a struct wouldn't we get an error?

Reply:
You should be using the struct which has the field right? How come you don't see the struct field?

Reply:
@whynowy you are right. the struct is imported from go.mod. @ayildirim21 we should always have access to the struct. could you update?

Reply:
ok yeah that makes sense since numaflow is using latest version of numaflow-go it will always have that field, we can simply the check to just seeing if it is empty string in that case